cmd/backup-manager/app/backup/backup.go

```diff
@@ -0,0 +1,224 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backup
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/mholt/archiver"
+	"github.com/pingcap/tidb-operator/cmd/backup-manager/app/constants"
+	"github.com/pingcap/tidb-operator/cmd/backup-manager/app/util"
+)
+
+// BackupOpts contains the input arguments to the backup command
+type BackupOpts struct {
+	Namespace   string
+	TcName      string
+	TidbSvc     string
+	Password    string
+	User        string
+	StorageType string
+	BackupName  string
+}
+
+func (bo *BackupOpts) String() string {
+	return fmt.Sprintf("%s/%s", bo.Namespace, bo.TcName)
+}
+
+func (bo *BackupOpts) getBackupFullPath() string {
+	return filepath.Join(constants.BackupRootPath, bo.getBackupRelativePath())
+}
+
+func (bo *BackupOpts) getBackupRelativePath() string {
+	BackupName := fmt.Sprintf("backup-%s", time.Now().UTC().Format(time.RFC3339))
+	return fmt.Sprintf("%s_%s/%s", bo.Namespace, bo.TcName, BackupName)
+}
+
+func (bo *BackupOpts) getDestBucketURI() string {
+	return fmt.Sprint("%s://%s%s", bo.StorageType, bo.getBackupRelativePath(), constants.DefaultArchiveExtention)
+}
+
+func (bo *BackupOpts) getTikvGCLifeTime(db *sql.DB) (string, error) {
+	var tikvGCTime string
+	row := db.QueryRow("select variable_value from ? where variable_name= ?", constants.TidbMetaTable, constants.TikvGCVariable)
+	err := row.Scan(&tikvGCTime)
+	if err != nil {
+		return tikvGCTime, fmt.Errorf("query cluster %s %s failed, err: %v", bo, constants.TikvGCVariable, err)
+	}
+	return tikvGCTime, nil
+}
+
+func (bo *BackupOpts) setTikvGClifeTime(db *sql.DB, gcTime string) error {
+	_, err := db.Exec("update ? set variable_value = ? where variable_name = ?", constants.TidbMetaTable, gcTime, constants.TikvGCVariable)
+	if err != nil {
+		return fmt.Errorf("set cluster %s %s failed, err: %v", bo, constants.TikvGCVariable, err)
+	}
+	return nil
+}
+
+func (bo *BackupOpts) dumpTidbClusterData() (string, error) {
+	bfPath := bo.getBackupFullPath()
+	err := util.EnsureDirectoryExist(bfPath)
+	if err != nil {
+		return "", err
+	}
+	args := []string{
+		fmt.Sprintf("--outputdir=%s", bfPath),
+		fmt.Sprintf("--host=%s", bo.TidbSvc),
+		"--port=4000",
+		fmt.Sprintf("--User=%s", bo.User),
+		fmt.Sprintf("--Password=%s", bo.Password),
+		"--long-query-guard=3600",
+		"--tidb-force-priority=LOW_PRIORITY",
+		"--verbose=3",
+	}
+
+	dumper := exec.Command("/mydumper", args...)
```

Comment:
why not use `exec.Command().CombinedOutput()`, then you can get the output when failed.

Reply:
On the one hand, it is divided into multiple steps to make the error finer. On the other hand, it is possible to introduce streaming backup data in the future. This requires exporting the stdout of the command, refer to [this](https://golang.org/pkg/os/exec/#Cmd.StdoutPipe).